### PR TITLE
Add structural schemas to CRDs

### PR DESCRIPTION
**Pull request by jasonwzm**
Please provide a description for what this PR is for.
Add generated structural schemas to CRDs

Starting k8s 1.15+, CRDs accept structural schemas for validation purposes. Design doc: https://docs.google.com/document/d/1ugwCNgQa4oxZK1JZ8NiaV7PAh4rI7H5MNwoF6LgOZIE/edit#heading=h.9t5cvudb0ru7

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[X] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: bin/yamllintrules [hunk 1]
```diff
@@ -5,6 +5,7 @@ ignore: |
   /tests/e2e/tests/pilot/testdata/networking/v1alpha3/rule-503test-destinationrule-c-add-subset.yaml
   /tests/e2e/tests/pilot/testdata/networking/v1alpha3/rule-503test-destinationrule-c-del-subset.yaml
   /tests/e2e/tests/pilot/testdata/networking/v1alpha3/destination-rule-c-private.yaml
+  /install/kubernetes/helm/istio-init/files/crd*.yaml
```
Comment (howardjohn) [inline bin/yamllintrules]: can you either remove from this list or justify why it is here?

Comment (jasonwzm) [reply bin/yamllintrules]: yes, this is a temp line here. i will take care of it once it is ready to review.

**Review by geeknoid — APPROVED** [review]

File: install/kubernetes/helm/istio-init/files/crd-14.yaml [hunk 2]
```diff
@@ -1,26 +1,141 @@
-kind: CustomResourceDefinition
 apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
 metadata:
-  name: authorizationpolicies.security.istio.io
+  creationTimestamp: null
```
Comment (howardjohn) [inline install/kubernetes/helm/istio-init/files/crd-14.yaml]:
```suggestion
```

File: install/kubernetes/helm/istio-init/files/crd-14.yaml [hunk 3]
```diff
@@ -1,26 +1,141 @@
-kind: CustomResourceDefinition
 apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
```
Comment (howardjohn) [inline install/kubernetes/helm/istio-init/files/crd-14.yaml]: nit: for clean history can you only modify what is needed

File: galley/testdata/validation/dataset/config-v1alpha2-QuotaSpec-valid.yaml [hunk 4]
```diff
@@ -9,5 +9,5 @@ spec:
         api.operation:
           exact: getProducts
     quotas:
-    - charge: 1
+    - charge: "1"
```
Comment (howardjohn) [inline galley/testdata/validation/dataset/config-v1alpha2-QuotaSpec-valid.yaml]: this should be an int?

File: install/kubernetes/helm/istio-init/files/crd-10.yaml [hunk 5]
```diff
@@ -1,560 +1,4813 @@
...
-    description: The destination hosts to which traffic is being sent
-    name: Hosts
-    type: string
-  - JSONPath: .metadata.creationTimestamp
-    description: |-
-      CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+  - name: v1alpha2
+    served: true
+    storage: true
+status:
```
Comment (howardjohn) [inline install/kubernetes/helm/istio-init/files/crd-10.yaml]: I don't think you can set the status, kubernetes will just ignore this?

File: install/kubernetes/helm/istio-init/files/crd-14.yaml (hunk 3 above)
Comment (jasonwzm) [reply install/kubernetes/helm/istio-init/files/crd-14.yaml]: This is a generated file. So I believe the generated yaml respects the order of fields in k8s api.

File: install/kubernetes/helm/istio-init/files/crd-10.yaml (hunk 5 above)
Comment (jasonwzm) [reply install/kubernetes/helm/istio-init/files/crd-10.yaml]: Yes, Kubernetes will just ignore this. The `status` field is the output of the crd generation.

Comment (howardjohn) [reply install/kubernetes/helm/istio-init/files/crd-10.yaml]: Is it feasible to not have the output generate this? If it is impossible thats fine but would prefer to not have random junk around, its confusing

Comment (jasonwzm) [reply install/kubernetes/helm/istio-init/files/crd-10.yaml]: It should be feasible. I can do that as a follow up to the generation tool.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**